text/0000-ember-data-packages.md

```diff
@@ -0,0 +1,396 @@
+      <td>none</td>
+    </tr>
+  </tbody>
+</table>
+
+### Notes
+
+#### `@ember-data/model`
+  
+  1) `InternalModel` and `RootState` are tightly coupled to the store and to our provided `Model`
+    implementation. Overtime we need to uncouple this, but given their coupling to `Model` and our
+    desire to enable them to be eliminated from projects not using `Model`, I believe these exports
+    belong in `@ember-data/model` for this reason.
+
+  2) Do the following belong in `@ember-data/model`, or in `@ember-data/relationship-layer`?
+    if not with relationships, does the package name "relationship-layer" become confusing?
+    
+   The argument for them being here is they are far more related to the current `Model` ui
+    presentation layer than they are to the state book-keeping which could be used with
+    alternative implementations.
+
+  * `belongsTo`
+  * `hasMany`
+  * `PromiseManyArray`
+  * `ManyArray`
+  
+#### `@ember-data/serializers`
+
+  1) We should move automatic registration of transforms into a more traditional
+    `app/` directory re-export for the package so that when the package is dropped they
+    cleanly drop as well.
+
+#### `@ember-data/store`
+
+  1) _setupContainer registers various adapters and serializers for fallbacks.
+  Either we need to deprecate this behavior (preferred), or separate out initialization
+  by package, or both. It also eagerly injects store, which we should deprecate but
+  can't until there is a `defaultStore` RFC for ember itself.
+
+  2) _initializeStoreService eagerly instantiates the store to ensure that `defaultStore` is our store.
+  we should get rid of this but can't until there is a `defaultStore` RFC for ember itself.
+
+  3) normalizeModelName is defined... very oddly. Why? Also we should probably deprecate this
+   and continue to move to a world in which less normalization of modelName is required.
+
+#### `@ember-data/relationship-layer`
+
+This package seems thin but it's likely to hold quite a bit.
+  Additional private things that would be moved here:
+  
+  * everything in `-private/system/relationships/state`
+  * `BelongsToReference` and `HasManyReference`
+  * relationship logic from `store` / `internal-model` that need to be isolated and extracted
+
+#### `@ember-data/debug`
+
+  Moving `DebugAdapter` here would allow dropping it if not desired. I already suspect we should
+  RFC dropping it for production builds where it adds persistent unnecessary overhead for a tool
+  meant for devs. This exists to support the ember inspector.
+
+### Documented Public APIs without public import paths
+
+There are a few public classes that are not exposed at all via `export` today. Those classes will not be given
+  public export paths, but the package containing their documentation and implementation is shown here:
+
+  * `@ember-data/store`
+    * `Reference`
+    * `RecordReference`
+  * `@ember-data/relationship-layer`
+    * `BelongsToReference`
+    * `HasManyReference`
```

Comment:
are these public in the sense that instances of these classes are exposed to end users? Or that they are meant to be imported and instantiated as standalone classes? If the former, I'm not sure a public export is required. 

For the record, the only time I've cared about these classes is to call `.load()` on them, and since there network fetch path is different from using `.get('relationshipName')`, I try not to use this API anymore.

Reply:
They are public in that we hand users instances of these classes, but they need no public import paths as users should not be instantiating them on their own. The documentation however ought to live alongside the appropriate mental model (and be where the code should live too)